src/pybind/mgr/cephadm/module.py

```diff
@@ -2535,10 +2535,42 @@ def _apply(self, spec: GenericSpec) -> str:
 
         return self._apply_service_spec(cast(ServiceSpec, spec))
 
+    def _get_candidate_hosts(self, placement: PlacementSpec) -> List[str]:
+        """Return a list of candidate hosts according to the placement specification."""
+        all_hosts = self.cache.get_schedulable_hosts()
+        draining_hosts = [dh.hostname for dh in self.cache.get_draining_hosts()]
+        ls = []
```

Comment:
I know this was taken from elsewhere, but since we're now doing it here outside the scheduler I think we should probably come up with a more descriptive variable name than `ls`

Reply:
fixed